### PR TITLE
Warning when Search Terms page is opened by clicking option at the footer 

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. magento/magento2#25245 Warning when Search Terms page is opened by clicking option at the footer 

### Manual testing scenarios (*)
1. Setup vanilla 2.3-develop
2. Create a product in the backend, give it the name "test"
3. On the frontend, perform a search query, search for "test"
4. Notice that it finds the product
5. If you now go to the Search Terms page in the frontend, it will show the search term "test" => **good**
6. Now in the backend, go to Marketing > SEO & Search > Search Terms
7. Open your "test" term
8. Set the "Number of Uses" to 0 and save
9. Click On Search terms option present at the bottom

**1. Page opens without any error
2. There is no errors in the system log** 
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by VladimirZaets** [comment]
Hi @vishalverma279.
Thanks for collaboration.
The processing of this PR will be after the issue will be confirmed

**Comment by ghost** [comment]
@VladimirZaets please check as the issue has been verified: #25245

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @vishalverma279,
Here is still an issue that variable `$termKeys` should be initialized with an empty array. Could you update your PR accordingly?

File: app/code/Magento/Search/Block/Term.php [hunk 1]
```diff
@@ -99,6 +100,9 @@ protected function _loadTerms()
                 $temp[$term->getQueryText()] = $term;
                 $termKeys[] = $term->getQueryText();
             }
+            if (empty($termKeys)) {
+                return $this;
+            }
```
Comment (ihor-sviziev) [inline app/code/Magento/Search/Block/Term.php]: Looks like we don't need this section anymore.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6842](https://jira.corp.magento.com/browse/ENGCOM-6842) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Before:**
before

**After:**
after

**Pull request merged**